src/test/java/com/gluonhq/richtext/undo/CmdManagerTests.java

```diff
@@ -52,6 +55,112 @@ public void redoRules() {
+        commander.execute(new TestCommand());
+        commander.undo();
+        Assertions.assertTrue(commander.isUndoStackEmpty());
+    }
+
+    @Test
+    @DisplayName("redo stack must not be empty after undo operation")
+    public void redoStackNotEmptyAfterUndoOperation() {
+        StringBuilder text = new StringBuilder("Text");
+        CommandManager<StringBuilder> commander = new CommandManager<>(text);
+        commander.execute(new TestCommand());
+        commander.undo();
+        Assertions.assertFalse(commander.isRedoStackEmpty());
+    }
+
+    @Test
+    @DisplayName("redo stack must be empty after redo operation")
+    public void redoStackEmptyAfterRedoOperation() {
+        StringBuilder text = new StringBuilder("Text");
+        CommandManager<StringBuilder> commander = new CommandManager<>(text);
+        commander.execute(new TestCommand());
+        commander.undo();
+        commander.redo();
+        Assertions.assertTrue(commander.isRedoStackEmpty());
+    }
+
+    @Test
+    @DisplayName("undo stack must be not empty after redo operation")
+    public void undoStackNotEmptyAfterRedoOperation() {
+        StringBuilder text = new StringBuilder("Text");
+        CommandManager<StringBuilder> commander = new CommandManager<>(text);
+        commander.execute(new TestCommand());
+        commander.undo();
+        commander.redo();
+        Assertions.assertFalse(commander.isUndoStackEmpty());
+    }
+
+    @Test
+    @DisplayName("runnable called when command executed")
+    public void runnableIsCalledWhenCommandIsExecuted() {
+        StringBuilder text = new StringBuilder("Text");
+        AtomicBoolean aBoolean = new AtomicBoolean();
+        CommandManager<StringBuilder> commander = new CommandManager<>(text, () -> aBoolean.set(!aBoolean.get()));
+        commander.execute(new TestCommand());
+        Assertions.assertTrue(aBoolean.get());
+    }
+
+    @Test
+    @DisplayName("runnable called when undo is executed")
+    public void runnableIsCalledWhenUndoIsExecuted() {
+        StringBuilder text = new StringBuilder("Text");
+        AtomicBoolean aBoolean = new AtomicBoolean();
+        CommandManager<StringBuilder> commander = new CommandManager<>(text, () -> aBoolean.set(!aBoolean.get()));
+        commander.execute(new TestCommand());
+        commander.undo();
+        Assertions.assertFalse(aBoolean.get());
+    }
+
+    @Test
+    @DisplayName("runnable called when commands are executed")
+    public void runnableIsCalledWhenCommandsAreExecuted() {
+        StringBuilder text = new StringBuilder("Text");
+        AtomicInteger aInteger = new AtomicInteger();
+        CommandManager<StringBuilder> commander = new CommandManager<>(text, aInteger::incrementAndGet);
+        Assertions.assertEquals(0, aInteger.get());
+        commander.execute(new TestCommand());
+        Assertions.assertEquals(1, aInteger.get());
+        commander.undo();
+        Assertions.assertEquals(2, aInteger.get());
```

Comment:
Nitpicking here, but Assertion of value 1 and 2 needs to be a part of the two previous tests.  We should perform `command.execute`, `undo` and `redo`. Finally, assert the value of `aInteger` to `3`